lib/ixml.js

```diff
@@ -161,9 +162,23 @@ const iXmlNodePgmOpen = (xname, xlib, xfunc, xerror) => iXmlNodeOpen(I_XML_NODE_
           + I_XML_NODE_CLOSE;
 const iXmlNodePgmClose = () => I_XML_NODE_PGM_CLOSE;
 
-const iXmlNodeParmOpen = xio => iXmlNodeOpen(I_XML_NODE_PARM_OPEN)
-          + iXmlAttrDefault(I_XML_ATTR_KEY_IO, xio, I_XML_ATTR_VALUE_OPTIONAL)
-          + I_XML_NODE_CLOSE;
+const iXmlNodeParmOpen = (opt) => {
+  if (!(opt && typeof opt==='object')){
+    return iXmlNodeOpen(I_XML_NODE_PARM_OPEN)
+    + I_XML_NODE_CLOSE;
+  }
+
+  let opt_io = (opt.io) ? iXmlAttrDefault(I_XML_ATTR_KEY_IO,opt.io,I_XML_ATTR_VALUE_OPTIONAL) : '';
+  let opt_by = (opt.by) ? iXmlAttrDefault(I_XML_ATTR_KEY_BY,opt.by,I_XML_ATTR_VALUE_OPTIONAL) : '';
+
+  delete opt.by;
```

Comment:
Not sure why we need to delete (or why the linter isnt complaining, since it hates delete in other places). These properties will get cleaned up after the function completes, shouldn't have to manually delete them.